### PR TITLE
Code snippets for solid

### PR DESCRIPTION
**Pull request by mauriver21**
## What I did

- After mergin this PR, [Frontpage PR](https://github.com/storybookjs/frontpage/pull/490) can be merged.
- Added documentation code snippets for SolidJS, there are still pending some code snippets to add for:
-- Mocking GraphQL queries with MSW addon
-- Setup the testing addon for your framework
- Added the file `docs/sync.js` for synchronizing docs with `frontpage` repo due to `link-monorepo-docs` command is not working.
- Updated `/docs/contribute/new-snippets.md` with new steps for updating code snippets in `frontpage`, due to `link-monorepo-docs` is unusable.

## How to test

- Follow `/docs/contribute/new-snippets.md` guide to setup `frontpage` and the monorepo docs synchronization.

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: docs/sync.js
Comment (kylegach) [inline docs/sync.js:26]: I tested this with a relative path `../../frontpage` and it works fine. Suggest asking for a relative path, as the absolute path can be unwieldy and is seldom known off-hand.

File: docs/contribute/new-snippets.md
Comment (kylegach) [inline docs/contribute/new-snippets.md:30]: Probably worth pointing out that this is an ongoing process, so they should run it in a separate terminal window.

File: docs/snippets/solid/button-story-click-handler-args.js.mdx
Comment (kylegach) [inline docs/snippets/solid/button-story-click-handler-args.js.mdx:1]: For snippets that contain JSX, we should use `jsx` for the language.

(I realize this is also incorrect for our React—and likely other–snippets. ☹️)

File: docs/snippets/solid/component-story-custom-args-complex.js.mdx
Comment (kylegach) [inline docs/snippets/solid/component-story-custom-args-complex.js.mdx:33]: I'm not very familiar with SolidJS. Is there someone from the community that could review the SolidJS-specific parts of this work?

File: docs/snippets/solid/button-group-story.ts-4-9.mdx
Comment (jonniebigodes) [reply docs/snippets/solid/button-group-story.ts-4-9.mdx:17]: @kylegach, it's from the pr that Kasper pushed for the initial React snippets. I've already updated them on my end and the they should fixed shortly.

**Review by kylegach — APPROVED** [review]
Wow! 🙌  What a _huge_ effort. Thank you, @webblocksapp!

I left a few comments, but the main thought is that I'm not actually qualified to review the SolidJS-specific aspects of this work. I'm OK to ship this as-is and revise from there, but I advise finding another community member with SolidJS experience to review.

**Comment by kylegach** [comment]
Oh, and I'm also not sure about the CI failures. Perhaps merging (or rebasing onto) the latest `next` would resolve them?

**Comment by jonniebigodes** [comment]
@webblocksapp once again, thank you so much for taking the time and effort to get the documentation updated with this pull request. We appreciate. With that out of the way, I hope you don't mind, but we're going to hold off on merging this pull request until Storybook 7.0 is out, as that will give us some time to reach out to the community and enlist someone else to help you, and us in getting the documentation added, more even as Kyle Gach (the other reviewer) and I don't have the framework knowledge to better assist you in getting this one in. If you know someone highly knowledgeable of Solid and interested in collaborating with all of us and helping out, please reach out to him, and we can coordinate the work amongst ourselves and get this out when humanly possible. 

Will that work for you?

Looking forward to hearing from you and getting this updated and merged.

I hope you have a great day.

Stay safe

File: docs/sync.js
Comment (webblocksapp) [reply docs/sync.js:1]: Maybe I can do this improvement by converting the `sync.js` file into a storybook task.

File: docs/sync.js
Comment (webblocksapp) [reply docs/sync.js:26]: Perfect, I will update the command message for asking for a relative path and docs too.

File: docs/contribute/new-snippets.md
Comment (webblocksapp) [reply docs/contribute/new-snippets.md:30]: May be by converting the `sync.js` into a task will simplify more the docs.

File: docs/snippets/solid/component-story-custom-args-complex.js.mdx
Comment (webblocksapp) [reply docs/snippets/solid/component-story-custom-args-complex.js.mdx:33]: I can do the review of the code snippets on a separate sandbox project, so I can attach a link to this PR with the sample repo so you can test from those implementations. Let me know if this works for you.

**Comment by webblocksapp** [comment]
Hi @jonniebigodes  and @kylegach, sorry for answering so far. It was a busy week, but now I have time again to continue the contribution. 

Maybe I would recommend merging the current docs as soon as this PR is stable, so the SolidJS community could start providing earlier feedback about the code snippets. So I could continue refining the documentation through that feedback so you can tag me freely on issues related to Solid.

Also, this PR updates the documentation about how to add code snippets and visualize them on the `frontpage` repo, so It will facilitate a lot the contribution process for updating code snippets from a Storybook newcomer contributor. 

> ... until Storybook 7.0 is out, as that will give us some time to reach out to the community and enlist someone else to help you ...

Perhaps, to help you with the review of the code snippets, I can do a separate sandbox project which tests them. So I can attach a link to this PR with the sample repo so you can test from those implementations. If you want more hands, I think someone you may contact to find them from the Solid community is @davedbase (Community Leader of SolidJS), he is at `wg-solid` discord working group of Storybook.

> Oh, and I'm also not sure about the CI failures. Perhaps merging (or rebasing onto) the latest next would resolve them?

When I did the PR, I merged with `next` however the CI failures persisted. It's weird because I only added .md files and the `sync.js` which is a separate script file for updating docs. 

For now, I can continue advancing on the improvements/fixes I commented to @kylegach on this PR until I receive your feedback. Good day to all 🙂

**Comment by jonniebigodes** [comment]
@webblocksapp, thanks for following up with me 🙏 ! Regarding your feedback, having a sandbox looks like a fantastic idea. If you're able to get it, we'd appreciate it. Also, if you can drop it in the Discord Solid working group and ping @davedbase for visibility, I'd appreciate it. Will that work for you? Also, be on the lookout as the docs are getting updated almost daily to factor in some changes regarding 7.0, so you'll probably have some conflicts already. 

Hope you have a great day.

Stay safe

**Comment by webblocksapp** [comment]
Hi @jonniebigodes and @kylegach,

I hope this message finds you well. I wanted to bring your attention to some recent updates that I just pushed to the repository. Most of the code snippets have been thoroughly tested in a sandbox repository using Storybook, giving us greater confidence that these examples are working as intended. If you'd like to check it out, the sandbox repository can be found [here](https://github.com/webblocksapp/solidjs-sb-sandbox).

In addition, we've added a new command at yarn task for synchronizing docs, which will make it easier for developers to get involved and contribute to the frontpage docs. You can find the documentation for this command at docs/contribute/new-snippets.md.

I suggest that we consider this approach as a first iteration for the SolidJS documentation and seek feedback from the community to gather insights and make necessary improvements in the future. 

After this PR revision and merge, you can move forward with the [Frontpage PR](https://github.com/storybookjs/frontpage/pull/490).

Thank you for your time and consideration.
Good weekend!

File: docs/snippets/solid/list-story-with-unchecked-children.js.mdx
Comment (jonniebigodes) [inline docs/snippets/solid/list-story-with-unchecked-children.js.mdx:1]: @webblocksapp one question about this snippet and the TS variations. I was looking at this and compared it with your reproduction [here](https://github.com/webblocksapp/solidjs-sb-sandbox/blob/master/src/stories/ListItemSelections.stories.tsx), and they don't match, to give you a bit more context that snippet is only featured for React as it's possible to apply this pattern (or until recently it did), with that I was wondering if you could update the reproduction and see if you apply this with Solid, otherwise, if not possible, we can leave the snippet in but won't reference any of them. Let me know, and we'll go from there.

File: docs/snippets/solid/mdx-canvas-multiple-stories.mdx.mdx
Comment (jonniebigodes) [inline docs/snippets/solid/mdx-canvas-multiple-stories.mdx.mdx:1]: @webblocksapp if you're ok with it, could you remove this snippet, as with Storybook 7 this pattern shouldn't be used. And other examples are now removed from our documentation

File: docs/snippets/solid/storybook-preview-global-decorator.js.mdx
Comment (jonniebigodes) [inline docs/snippets/solid/storybook-preview-global-decorator.js.mdx:2]: @webblocksapp, to prevent the snippets from reaching a point they can become hard to maintain, could you apply the same pattern used for React [here](https://github.com/storybookjs/storybook/blob/next/docs/snippets/react/storybook-preview-global-decorator.js.mdx) and condense the decorator's snippets into a single one for JavaScript and one for TypeScript and remove the extra ones. As you already applied this to some extent but forgot to update the remainder examples.

File: docs/snippets/solid/your-component-with-decorator.js.mdx
Comment (jonniebigodes) [inline docs/snippets/solid/your-component-with-decorator.js.mdx:1]: If you could apply the same pattern here as to what was mentioned for the `storybook-preview-global-decorator.js.mdx` I'd appreciate it.

File: docs/writing-stories/introduction.md
Comment (jonniebigodes) [inline docs/writing-stories/introduction.md:91]: Just a small item here, if you could leave in the dangling comma, I'd appreciate it. It's just for consistency's sake.

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@webblocksapp Sorry for the wait on this, but as you may have heard, Storybook 7 is out and now I've had some time to review this pull request in more detail and with that I left some small items for you to address so that we can merge this and the companion pull request and start documenting the SolidJS framework.


Let me know once you've addressed this and we'll go from there.

Hope you have a great weekend.

Stay safe

File: docs/writing-stories/introduction.md
Comment (webblocksapp) [reply docs/writing-stories/introduction.md:91]: Fixed.

File: docs/snippets/solid/storybook-preview-global-decorator.js.mdx
Comment (webblocksapp) [reply docs/snippets/solid/storybook-preview-global-decorator.js.mdx:2]: Fixed.

File: docs/snippets/solid/your-component-with-decorator.js.mdx
Comment (webblocksapp) [reply docs/snippets/solid/your-component-with-decorator.js.mdx:1]: Fixed.

File: docs/snippets/solid/mdx-canvas-multiple-stories.mdx.mdx
Comment (webblocksapp) [reply docs/snippets/solid/mdx-canvas-multiple-stories.mdx.mdx:1]: Done.

File: docs/snippets/solid/list-story-with-unchecked-children.js.mdx
Comment (webblocksapp) [reply docs/snippets/solid/list-story-with-unchecked-children.js.mdx:1]: I'ven't figured out that scenario, I've just added it inside the solid sb sandbox repo ([here](https://github.com/webblocksapp/solidjs-sb-sandbox/blob/master/src/stories/ListStoryUncheckedChildren.stories.tsx)) and it's not supported. So I've removed the references of any of them in the docs.

File: docs/snippets/solid/component-story-static-asset-cdn.js.mdx
Comment (webblocksapp) [reply docs/snippets/solid/component-story-static-asset-cdn.js.mdx:2]: Fixed.

File: docs/snippets/solid/component-story-figma-integration.ts-4-9.mdx
Comment (webblocksapp) [reply docs/snippets/solid/component-story-figma-integration.ts-4-9.mdx:10]: Removed comment only for Solid code snippets. 

This comment is also present in the other frameworks.

File: docs/snippets/solid/button-story-default-export-with-component.ts-4-9.mdx
Comment (webblocksapp) [reply docs/snippets/solid/button-story-default-export-with-component.ts-4-9.mdx:2]: Fixed.

File: docs/snippets/solid/button-story-decorator.ts.mdx
Comment (webblocksapp) [reply docs/snippets/solid/button-story-decorator.ts.mdx:1]: Fixed.

File: docs/snippets/solid/button-story-auto-docs.ts.mdx
Comment (webblocksapp) [reply docs/snippets/solid/button-story-auto-docs.ts.mdx:2]: Fixed.

**Comment by webblocksapp** [comment]
Hi @jonniebigodes, I wanted to let you know that I've addressed the feedback from the code review. Feel free to let me know if you have some other findings. Thanks!

**Review by shilman — COMMENTED** [review]
Hi @webblocksapp !! I tried out the sync-docs task. It's a really cool idea.

I had a couple minor comments about the experience:

1. I ran it on an unsynced `frontpage` project. It errored because the `src/content/docs` directory didn't exist yet. If the `src/content` directory exists but there is no `docs` directory, it should create a `docs` directory for you.
2. I manually created the `docs` directory. However it was empty. Then every time I updated the monorepo it copied the changes over to the new `docs` directory. However, the rest of the content was not copied over, so the workflow wasn't really working. If you are creating the the `docs` directory per the previous point, maybe the task should copy over the entire contents on initialization?

Another thought is... would it be simpler to just use symbolic links instead & update the documentation accordingly, rather than creating a new task?

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@webblocksapp, thanks for addressing the feedback so promptly. Appreciate it 🙏 ! From my end, I just need for you to look into a small item regarding the ts-4-9 variants of the snippets, you already addressed it in some, but I was wondering if you could take another pass at them and adjust accordingly. Also, when you have a chance, can you take a look at the feedback from @shilman so that we can get this one merged and update the documentation? 

Looking forward to hearing from you. 

Hope you have a great week.

**Comment by shilman** [comment]
@webblocksapp your solution sounds great to me. thanks for taking care of this!! 🙏

File: docs/snippets/solid/button-group-story.ts-4-9.mdx
Comment (webblocksapp) [reply docs/snippets/solid/button-group-story.ts-4-9.mdx]: Done!

**Comment by webblocksapp** [comment]
Hello @jonniebigodes and @shilman, I wanted to let you know that I've just pushed new updates to the code based on your feedback. Please take a look and let me know if there's anything else that needs to be refined. Thank you!

**Review by jonniebigodes — APPROVED** [review]

**Comment by jonniebigodes** [comment]
@webblocksapp I've just checked, and all is good on my end. Thanks once again for addressing the feedback so promptly. Appreciate it 🙏 ! I'll check with the team, and if there's nothing blocking it, we can finally merge this one and the associated pull request in the frontpage repository to get the documentation updated and improve the UX for contributions. 

Hope you have a fantastic day.

Stay safe

**Review by shilman — APPROVED** [review]
Thanks for addressing my feedback @webblocksapp !! Amazing job on this PR 👏

**Comment by jonniebigodes** [comment]
@webblocksapp going to merge this. Once again, thanks for the amazing job with the contribution 🎉

**Pull request merged**
